gateway-common/src/main/java/com/ca/mfaas/product/routing/transform/TransformService.java

```diff
@@ -50,8 +52,14 @@ public String transformURL(ServiceType type,
             throw new URLTransformationException(message);
         }
 
-        String path = serviceUri.getPath().replace(route.getServiceUrl(), "");
-        if (!path.isEmpty() && !path.startsWith("/")) {
+
+        String pathToReplace = serviceUri.getPath();
```

Comment:
It's looks good, but only one note:
I didn't found in this method check on `serviceUri.getPath() != null`, because if `serviceUrl` will be incorrect format, for example, like that: `https:locahost:8080/`, we will catch NPE. I don't know may be this situation can't be happen at all, but if can be, I think will be better to add this check just right before 48 line.